### ISSUE TITLE
Should we install Deepspeed or not?

### ISSUE THREAD

**Issue by runner22k** [bmaltais/kohya_ss#42]
### For anyone who is trying to install deepspeed. Should we install Deepspeed or not?

While installing when we execute **accelerate config** we are asked to install deepspeed or not. If you select yes, we will be asked to run below pip. That is when I am getting this error.

Luckily I opened https://github.com/kohya-ss/sd-scripts

there developer mentioned answers too. Where we are asked to not install deepspeed. 

### Steps to reproduce the problem

1. In the VENV run **accelerate config**.
2. When asked to install deepspeed select YES.
3. Run **pip install deepspeed** 

### What should have happened?

deepshpeed should be installed. But we are getting CUDA error.

### What platforms do you use to access UI ?

Windows

### What browsers do you use to access the UI ?

Mozilla Firefox

### Command Line Arguments

```Shell
pip install deepspeed
```

### Additional information, context and logs

```
(venv) PS D:\kohya_ss> pip install deepspeed
Collecting deepspeed
  Using cached deepspeed-0.7.7.tar.gz (712 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\jan\AppData\Local\Temp\pip-install-bh_2b2bj\deepspeed_337b9c20b5434938ad45f7a28ad99337\setup.py", line 78, in <module>
          cupy = f"cupy-cuda{''.join(map(str,installed_cuda_version()))}"
        File "C:\Users\jan\AppData\Local\Temp\pip-install-bh_2b2bj\deepspeed_337b9c20b5434938ad45f7a28ad99337\op_builder\builder.py", line 41, in installed_cuda_version
          assert cuda_home is not None, "CUDA_HOME does not exist, unable to compile CUDA op(s)"
      AssertionError: CUDA_HOME does not exist, unable to compile CUDA op(s)
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
(venv) PS D:\kohya_ss>
```

**Comment by bmaltais** [comment]
No need to... but it should be hurt.

**Comment by Odawgthat** [comment]
I got this error too, what should I do???

**Comment by Odawgthat** [comment]
> No ne

What does that mean

**Comment by bmaltais** [comment]
No idea. I suggest you raise this directly with kohya_ss since he create the actual training code. I only create the GUI around them to make easier for users.

**Comment by runner22k** [comment]
@Odawgthat I switched to Linux and google Colab, as we can't do much without deepspeed on 8GB or less VRAM GPUs. It is available only on Linux, and it will help us run some trainings. But definitely it will take much much longer. I could have spent more money on GPU with 16GB VRAM instead of buying 64GB system RAM. But its too late now. I realized the mistake.

I lost interest to train on my system. Maybe in the future I will upgrade to a new GPU with more VRAM and Linux as base, as linux has more work around comparatively.
